packages/experiments/src/components/Text/Text.tsx

```diff
@@ -44,17 +51,39 @@ const view = (props: IViewProps<ITextProps, ITextStyles>) => {
     wrap,
     ...rest
   } = props;
-
-  return <RootType {...rest} className={classNames.root} />;
+  return <RootType {...rest} className={props.styles.root} />;
 };
 
 const styles = (props: IStyleProps<ITextProps, ITextStyles>): ITextStyles => {
-  const { block, theme, wrap, grow, shrink, type, family, weight, size } = props;
-  const { typography } = theme;
-  const themeType = typography.types[type!] || {
-    fontFamily: typography.families[family!] || typography.families.default,
-    fontWeight: typography.weights[weight!] || typography.weights.default,
-    fontSize: typography.sizes[size!] || typography.sizes.medium
+  const { block, theme, wrap, grow, shrink, /* type, */ family, weight, size, style, color } = props;
+
+  const { palette, fonts, /* semanticColors, isInverted, */ typography } = theme;
+
+  const themeType = {
+    fontFamily: typography
```

Comment:
This block of logic isn't very readable, can you separate it out into chunks above or add comments?